ravedude/src/main.rs

```diff
@@ -91,12 +96,53 @@ fn main() {
 }
 
 fn ravedude() -> anyhow::Result<()> {
-    let args: Args = structopt::StructOpt::from_args();
+    let mut args: Args = structopt::StructOpt::from_args();
+    if args.dump_config {
+        println!(
+            "{}",
+            toml::to_string(&board::get_board(args.board.as_deref())?)?
+        );
+        return Ok(());
+    }
+
+    let ravedude_config_exists = std::path::Path::new("Ravedude.toml").try_exists()?;
```

Comment:
Right now you are searching for `Ravedude.toml` in the current working directory.  This breaks when you e.g. `cargo run` inside the `src/` directory.  Maybe you can check how other tools like `cargo-embed` find their config file and implement something similar?

As a hint, it seems cargo sets some environment variables when running a runner process.  One of them is `CARGO_MANIFEST_DIR` which seems to indicate the directory where cargo found its own `Cargo.toml`...

Reply:
`cargo-embed` uses [figment](https://docs.rs/figment/latest/figment/), which looks interesting. I'm most likely gonna switch ravedude over to it to mirror its behavior (and cargo's behavior) with environment variables, although I've held back on it for now due to it possibly being overkill (?) although ravedude itself isn't going to run on a microcontroller so that probably wasn't a good reason.

`CARGO_MANIFEST_DIR` seems to be only available when using ravedude as a runner. ravedude should be able to access Ravedude.toml no matter how it's being run, in my opinion.

Currently, I just have ravedude scan all parent directories until it finds a `Ravedude.toml` somewhere, which works fine for now... this behavior will probably be replaced by figment, though.